posts/2021-01-22-jwtbuilder-redirects-21001.adoc

```diff
@@ -0,0 +1,182 @@
+* <<bugs, Notable bugs fixed in this release>>
+
+// end::intro[]
+
+// tag::run[]
+[#run]
+
+== Run your apps using 21.0.0.1
+
+If you're using link:{url-prefix}/guides/maven-intro.html[Maven], here are the coordinates:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.openliberty</groupId>
+    <artifactId>openliberty-runtime</artifactId>
+    <version>21.0.0.1</version>
+    <type>zip</type>
+</dependency>
+----
+
+Or for link:{url-prefix}/guides/gradle-intro.html[Gradle]:
+
+[source,gradle]
+----
+dependencies {
+    libertyRuntime group: 'io.openliberty', name: 'openliberty-runtime', version: '[21.0.0.1,)'
+}
+----
+
+Or if you're using Docker:
+
+[source]
+----
+FROM open-liberty
+----
+//end::run[]
+
+Or take a look at our link:{url-prefix}/downloads/[Downloads page].
+
+[link=https://stackoverflow.com/tags/open-liberty]
+image::img/blog/blog_btn_stack.svg[Ask a question on Stack Overflow, align="center"]
+
+//tag::features[]
+
+[#jwtbuilder]
+=== Scheduled payloads with JWT Builder
+
+New in Open Liberty 21.0.0.1, the `jwtBuilder` element has been enhanced with a new attribute called `nbfOffset` which can be used to configure an NBF claim for a JWT payload. The time set for the `nbfOffset` will be added to the current time and the result will determine when Json Web Tokens will start to be accepted for processing.
+
+To configure the "not-before" claim using `jwtBuilder`, add the following to your `sever.xml` configuration file.
+
+`<jwtBuilder nbfOffset="1800s" />`
+
+If the JWT was issued at the current time, then the token can only be used after 1800 seconds have passed from the current time.
+
+`jwtBuilder` is a part of the `jwt-1.0` feature, to add the feature to your project add the following to the `server.xml`:
+
+[source, xml]
+----
+<server>
+  <featureManager>
+    <feature>jwt-1.0</feature>
+  </featureManager>
+</server>
+----
+
+For more information:
+
+* link:https://openliberty.io/docs/21.0.0.1/reference/config/jwtBuilder.html[Open Liberty JWT Builder Documentation]
+
+* link:https://www.ibm.com/support/knowledgecenter/SSAW57_liberty/com.ibm.websphere.wlp.nd.multiplatform.doc/ae/twlp_sec_build_jwt.html[Building JSON Web Tokens in Liberty]
+
+[#redirects]
+== Relative linking for HTTP Redirects
+
+Included in Open Liberty 21.0.0.1 is an enhancement for the `servlet-4.0` feature, within Open Liberty the `sendRedirect()` is used to direct a client to a new page or location away from the original page. Previously, Open Liberty would always convert the provided relative URL in the `sendRedirect()` function to an absolute URL. This could lead to problems for applications that took advantage of reverse proxy servers.
+
+To solve the problems presented, Open Liberty 21.0.0.1 ahs introduced a new `redirecttorelativeurl` web container property that will tell the application whether or not to construct absoulte URLs from relative redirect links. This property can be set in the `server.xml` file:
```

Comment:
Confirm that `redirecttorelativeurl` is correct

Reply:
Taken directly from https://github.com/OpenLiberty/open-liberty/issues/15605 @pmd1nh can you confirm this is correct?

Reply:
`redirecttorelativeurl` is correct.

Reply:
`will tell the application whether or not`  should change to `will tell the server whether or not`

Reply:
it is the server code that will take the relative URL which passes in by the application via sendRedirect() and convert to either an absolute or relative URL, depending on the config `redirecttorelativeurl` flag